CMakeLists.txt

```diff
@@ -1,93 +1,41 @@
-######### CMake Version #####################
-cmake_minimum_required(VERSION 2.8.11)
-#############################################
+project(eigen-cuda LANGUAGES CXX)
+cmake_minimum_required(VERSION 3.14)
 
-######### Options ###########################
-option( CORE_USE_CUDA    "Use CUDA to speed up certain parts of the code."              ON )
-#############################################
+option(USE_CUDA "Whether to use CUDA" OFF)
 
-
-######### CUDA decisions ####################
-if (CORE_USE_CUDA)
-    MESSAGE( STATUS ">> --------------  USING CUDA --------------" )
-    set( CUDA_TOOLKIT_ROOT_DIR "/opt/cuda" )
-    if (APPLE OR UNIX)
-        set(CMAKE_C_COMPILER   /opt/cuda/bin/gcc)
-        set(CMAKE_CXX_COMPILER /opt/cuda/bin/g++)
-    elseif (WIN32)
-        ### By default we use VS
-        MESSAGE( STATUS ">> User compiler:             MSVC" )
-        MESSAGE( STATUS ">> Choosing a different compiler is not yet implemented for Windows" )
-    endif()
-endif()
-#############################################
-
-
-######### Info ##############################
-MESSAGE( STATUS ">> CMAKE_C_COMPILER:        " ${CMAKE_C_COMPILER} )
-MESSAGE( STATUS ">> CMAKE_CXX_COMPILER:      " ${CMAKE_CXX_COMPILER} )
-#############################################
-
-
-######### Project Name ######################
-project(eigencuda)
-SET( EXECUTABLE_NAME run )
-#############################################
-
-
-### Find includes in corresponding build directories
-set( CMAKE_INCLUDE_CURRENT_DIR ON )
-######### Have the binary placed into the source head
-set( CMAKE_RUNTIME_OUTPUT_DIRECTORY ${CMAKE_SOURCE_DIR} )
-### Output paths for multi-config builds (e.g. msvc)
-foreach( OUTPUTCONFIG ${CMAKE_CONFIGURATION_TYPES} )
-    string( TOUPPER ${OUTPUTCONFIG} OUTPUTCONFIG )
-    set( CMAKE_RUNTIME_OUTPUT_DIRECTORY_${OUTPUTCONFIG} ${CMAKE_SOURCE_DIR} )
-    # set( CMAKE_LIBRARY_OUTPUT_DIRECTORY_${OUTPUTCONFIG} ${youroutputdirectory} )
-    # set( CMAKE_ARCHIVE_OUTPUT_DIRECTORY_${OUTPUTCONFIG} ${youroutputdirectory} )
-endforeach( OUTPUTCONFIG CMAKE_CONFIGURATION_TYPES )
-#############################################
-set( CMAKE_DISABLE_SOURCE_CHANGES  ON )
-set( CMAKE_DISABLE_IN_SOURCE_BUILD ON )
-#############################################
-
-
-######### CUDA decisions ####################
-if (CORE_USE_CUDA)
-    find_package(CUDA REQUIRED)
+if (USE_CUDA)
+    enable_language(CUDA)
     add_definitions(-DUSE_CUDA)
```

Comment:
if you'd like to modernise the cmake here, you could place this further below with the `set_property`:
```
 target_compile_definitions(test PRIVATE
        USE_CUDA
    )
```

(also, I believe this `if (USE_CUDA)` could be merged with the one further below?)

Reply:
added `USE_CUDA` as private property.

On the other note: The language has to be enabled or Eigen won't compile with cuda support. Not sure how to do this differently

Reply:
I'm not suggesting to remove it, but it seems like it should be fine to place
```
    enable_language(CUDA)
    add_definitions(-DUSE_CUDA)
```
into the if-block further down, 
```
if (USE_CUDA)
    add_library(test ........
```

Or do you mean `enable_language(CUDA)` needs to be called _before_ `FetchContent_MakeAvailable(Eigen3)`? If that's the case it's fine like it is